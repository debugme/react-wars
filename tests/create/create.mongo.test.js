import _ from 'lodash'
import path from 'path'
import dotenv from 'dotenv'
import mongoose from 'mongoose'

import schema from '../../schemas/CharacterSchema'
import fields from './create.data.json'

describe('create new document in characters collection', () => {

  let Character = null

  beforeAll((done) => {
    mongoose.Promise = global.Promise
    dotenv.config({ path: path.resolve(__dirname, '..', '.env') })
    const clearCollection = () => Character.collection.remove()
    Character = mongoose.model('Characters', schema)
    mongoose.connect(process.env.mongodbUri)
      .then(clearCollection)
      .then(done)
  })

  afterAll((done) => {
    Character.collection.remove()
      .then(() => mongoose.connection.close())
      .then(done)
  })

  it('should allow creation of a character from correct data', (done) => {
    const { name } = fields
    const keys = Object.keys(fields)
    const find = () => Character.findOne({ name })
    const check = (data) => expect(_.pick(data, keys)).toMatchSnapshot()
    new Character(fields).save()
      .then(find)
      .then(check)
      .then(done)
  })

  it('should disallow creation of a character from incorrect data', (done) => {
    const fields = { fruit: 'banana'}
    const check = (error) => expect(JSON.stringify(error)).toMatchSnapshot()
    new Character(fields).save()
      .catch(check)
      .then(done)
  })

})