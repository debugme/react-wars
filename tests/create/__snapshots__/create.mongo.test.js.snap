// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create new document in characters collection should allow creation of a character from correct data 1`] = `
Object {
  "birth_year": "29BBY",
  "eye_color": "brown",
  "hair_color": "brown",
  "height": 180,
  "is_male": true,
  "mass": 80,
  "name": "Han Solo",
  "skin_color": "fair",
}
`;

exports[`create new document in characters collection should disallow creation of a character from incorrect data 1`] = `"{\\"errors\\":{\\"is_male\\":{\\"message\\":\\"Path \`is_male\` is required.\\",\\"name\\":\\"ValidatorError\\",\\"properties\\":{\\"type\\":\\"required\\",\\"message\\":\\"Path \`{PATH}\` is required.\\",\\"path\\":\\"is_male\\"},\\"kind\\":\\"required\\",\\"path\\":\\"is_male\\"},\\"name\\":{\\"message\\":\\"Path \`name\` is required.\\",\\"name\\":\\"ValidatorError\\",\\"properties\\":{\\"type\\":\\"required\\",\\"message\\":\\"Path \`{PATH}\` is required.\\",\\"path\\":\\"name\\"},\\"kind\\":\\"required\\",\\"path\\":\\"name\\"}},\\"message\\":\\"Characters validation failed\\",\\"name\\":\\"ValidationError\\"}"`;
